part2_client.py

```diff
@@ -1,58 +1,48 @@
 # Group#: G37
 # Student Names: Ryley McRae & Niko Andrianos
 
+from socket import *
+import time
+
 """
-    This program implements a client for a UDP Pinger 
-    application on a local host.
+    Implements a client for a UDP Pinger application on a local host.
 """
+class Client:
+    def __init__(self, serverName: str = "localhost", serverPort: int = 12000, bufferSize: int = 2048):
+        self.serverName = serverName
+        self.serverPort = serverPort
+        self.bufferSize = bufferSize
+        self.clientSocket = socket(AF_INET, SOCK_DGRAM)
+        self.clientSocket.settimeout(1.0) # timeout exception will be raised 1 second after trying to receive data
+    
+    def close(self):
+        self.clientSocket.close()
+
+    def ping(self, request: str):
+        try: # use a try/except to handle a timeout from the server side
+            tripStartTime = time.time() # save the time of right before the request is made
+            self.clientSocket.sendto(message.encode(), (self.serverName, self.serverPort)) # send request to server
+            response, serverAddress = self.clientSocket.recvfrom(self.bufferSize) # fetch the response
+            roundTripTime = round(1000 * (time.time() - tripStartTime), 3) # calculate the round-trip-time in ms
+            print(f"Request:\t{request}\nResponse:\t{response.decode()}\nRound trip:\t{roundTripTime}ms")
+        except timeout:
+            print("Request timed out")
+        except Exception:
```

Comment:
I don't think the specifications on canvas had anything for handling an unknown exception. But, it isn't incorrect, seems like it could catch something wrong with sendto.

Reply:
This was just for debugging. It can be commented out. And yes you are right, - an unknown exception would be caught when the server wasn't running (raised by socket.sendto())